### PR TITLE
add valid user and admin cache

### PR DESCRIPTION
**Pull request by zhlu9890**

### CODE REVIEW

File: lib/SampleService/core/user_lookup.py [hunk 1]
```diff
@@ -91,12 +108,20 @@ def invalid_users(self, usernames: Sequence[UserID]) -> List[UserID]:
...
+
+        r = requests.get(self._user_url + ','.join([u.id for u in bad_usernames]),
                          headers={'Authorization': self._token})
         self._check_error(r)
         good_users = r.json()
+        for u in bad_usernames:
+            if u.id in good_users:
+                self._valid_cache.set(u.id, True)
+
         # TODO ACL cache
```
Comment (scanon) [inline lib/SampleService/core/user_lookup.py:123]: Drop the comment since this is adding the ACL cache

**Review by scanon — APPROVED** [review]
This looks good to me but I would like someone else to review it too.

File: lib/SampleService/core/user_lookup.py (hunk 1 above)
Comment (zhlu9890) [reply lib/SampleService/core/user_lookup.py]: Done

**Comment by slebras** [comment]
Look good to me also. It might be good to have a test that uses the functoins that use the caching and test to see if the user or admin roles are stored afterwards

**Comment by zhlu9890** [comment]
I didn't have test directly calling the cache store function, but I did have tests calling `valid_user` and `is_admin` functions which should cache previously checked user. After calling those two functions to check user, the user could be retrieved without calling remote service.

**Pull request merged**